HomeWork6/Khirniy_S_HW6.cpp

```diff
@@ -0,0 +1,165 @@
+// Khirniy.S HW6 - Linked list
+
+#include <iostream>
+
+template <typename T> class List {
+public:
+  List();
+  ~List();
+  void pop_front();
+  void push_back(T data);          // add the elements of the end of list
+  void clear();                    // method to clear the list
+  void push_front(T data);         // add the elements of the start of list
+  void insert(T value, int index); // insert element from index to list
+  void removeAt(int index);        // remove element from index
+  void pop_back();                 // remove element from end of list
+  int GetSize() { return Size; }   // method for return number of items
+  T &operator[](const int index);  // operator[] for return data from list
+
+private:
+  // class Node with data
+  template <typename T> class Node {
+  public:
+    Node *pNext;
+    T data;
+    Node(T data = T(), Node *pNext = nullptr) {
+      this->data = data;
+      this->pNext = pNext;
+    }
```

Comment:
а список инициализации зачем?
```suggestion
    Node(T data = T(), Node *pNext = nullptr) : data{data}, pNext{nullptr} {
    }
```